core/src/main/java/org/bitcoinj/wallet/AnyDeterministicKeyChain.java

```diff
@@ -233,6 +237,7 @@ private int calcDefaultLookaheadThreshold() {
         protected IDeterministicKey spendingKey = null;
         protected ImmutableList<ChildNumber> accountPath = null;
         protected KeyFactory keyFactory = ECKeyFactory.get();
+        protected boolean hardenedKeysOnly = false;
```

Comment:
For some purposes, we only want hardened keys or we can only create hardened keys (Ed25519 only supports hardened key derivation according to SLIP-0010).